program/rust/src/utils.rs

```diff
@@ -135,3 +145,102 @@ pub fn is_component_update(cmd_args: &UpdPriceArgs) -> Result<bool, OracleError>
         _ => Ok(false),
     }
 }
+
+
+/// These 3 accounts need to get passed to make sure that the upgrade authority is signing the
+/// transaction
+/// - `program_account` is the program at address `program_id`. It just contains a pointer to the
+///   `programdata_account`
+/// - `programdata_account` has an `upgrade_authority_address` field that needs to match
+///   `upgrade_authority.key`
+pub fn check_is_valid_upgradeauthority_program_programdata_triplet(
```

Comment:
typo: upgradeauthority

Reply:
you could also call this `check_is_upgrade_authority_for_program` . (I realize it does a bit more than that, but all of those checks are means to the end of checking the upgrade authority.)

Reply:
YEs